numba/parfors/parfor_lowering.py

```diff
@@ -1747,11 +1757,27 @@ def load_range(v):
                                                   context.get_constant(types.uintp, num_dim),
                                                   dim_starts,
                                                   dim_stops])
+    # Set the chunksize to zero so that any nested calls get the default chunk size behavior.
     builder.call(set_chunksize, [zero])
 
+    # Each entry in the schedule is 2 times the number of dimensions long.
     multiplier = context.get_constant(types.uintp, num_dim * 2)
+    # Compute the total number of entries in the schedule.
     sched_size = builder.mul(num_divisions, multiplier)
```

Comment:
Is there a statically known maximum for `num_divisions`?

If yes, it is possible to use `alloca_once(max_sched_size)` to avoid the malloc/free.

Reply:
With the chunking API, if you want, you can have each point in the iteration space be its own division of work and so the maximum divisions is bounded only by memory size.  The only way to be sure you'd never have to reallocate something larger is to allocate all of memory to the schedule.  This PR was designed for a possible underlying caching implementation so rather than allocate and deallocate each time, the deallocate routine can actually hold onto the memory and reuse it in the next allocation if it is no larger than the previous one.